source/opencascade/boundary_lib.cc

```diff
@@ -56,12 +59,12 @@ namespace OpenCASCADE
               (shape.ShapeType() == TopAbs_EDGE),
               ExcUnsupportedShape());
       if (shape.ShapeType() == TopAbs_WIRE)
-        return (Handle(BRepAdaptor_HCompCurve(new BRepAdaptor_HCompCurve(TopoDS::Wire(shape)))));
+        return Handle(BRepAdaptor_HCompCurve)(new BRepAdaptor_HCompCurve(TopoDS::Wire(shape)));
       else if (shape.ShapeType() == TopAbs_EDGE)
-        return (Handle(BRepAdaptor_HCurve(new BRepAdaptor_HCurve(TopoDS::Edge(shape)))));
+        return Handle(BRepAdaptor_HCurve)(new BRepAdaptor_HCurve(TopoDS::Edge(shape)));
 
       Assert(false, ExcInternalError());
-      return Handle(BRepAdaptor_HCurve(new BRepAdaptor_HCurve()));
+      return Handle(BRepAdaptor_HCurve)(new BRepAdaptor_HCurve());
```

Comment:
@luca-heltai It is fascinating how the old version (with the wrong parenthesis) worked by accident :-D